certify-service/Dockerfile

```diff
@@ -0,0 +1,94 @@
+ARG artifactory_url
+
+# environment variable to pass active profile such as DEV, QA etc at docker runtime
+ENV active_profile_env=${active_profile}
+
+# environment variable to pass github branch to pickup configuration from, at docker runtime
+ENV spring_config_label_env=${spring_config_label}
+
+# environment variable to pass spring configuration url, at docker runtime
+ENV spring_config_url_env=${spring_config_url}
+
+# environment variable to pass glowroot, at docker runtime
+ENV is_glowroot_env=${is_glowroot}
+
+# environment variable to pass artifactory url, at docker runtime
+ENV artifactory_url_env=${artifactory_url}
+
+# can be passed during Docker build as build time environment for github branch to pickup configuration from.
+ARG container_user=mosip
+
+# can be passed during Docker build as build time environment for github branch to pickup configuration from.
+ARG container_user_group=mosip
+
+# can be passed during Docker build as build time environment for github branch to pickup configuration from.
+ARG container_user_uid=1001
+
+# can be passed during Docker build as build time environment for github branch to pickup configuration from.
+ARG container_user_gid=1001
+
+
+# install packages and create user
+RUN apt-get -y update \
+&& apt-get install -y unzip file sudo \
+&& groupadd -g ${container_user_gid} ${container_user_group} \
+&& useradd -u ${container_user_uid} -g ${container_user_group} -s /bin/sh -m ${container_user} \
+&& id -u ${container_user} &>/dev/null || adduser ${container_user}
+
+RUN id -u ${container_user}
+# set working directory for the user
+WORKDIR /home/${container_user}
+
+ENV work_dir=/home/${container_user}
+
+ARG loader_path=${work_dir}/additional_jars/
+
+RUN mkdir -p ${loader_path}
+
+ENV loader_path_env=${loader_path}
+
+COPY ./target/certify-service-*.jar certify-service.jar
+
+# change permissions of file inside working dir
+RUN chown -R ${container_user}:${container_user} /home/${container_user}
+
+# select container user for all tasks
+USER ${container_user_uid}:${container_user_gid}
+
+EXPOSE 8090
+EXPOSE 9010
+
+#ENTRYPOINT [ "./configure_start.sh" ]
+CMD if [ "$is_glowroot_env" = "present" ]; then \
+    wget -q --show-progress "${artifactory_url_env}"/artifactory/libs-release-local/io/mosip/testing/glowroot.zip ; \
+    unzip glowroot.zip ; \
+    rm -rf glowroot.zip ; \
+    sed -i 's/<service_name>/idp-service/g' glowroot/glowroot.properties ; \
+    java -jar -javaagent:glowroot/glowroot.jar -Dloader.path="${loader_path_env}" -Dspring.cloud.config.label="${spring_config_label_env}" -Dspring.profiles.active="${active_profile_env}" -Dspring.cloud.config.uri="${spring_config_url_env}" certify-service.jar ; \
+    else \
+    java -jar -Dloader.path="${loader_path_env}" -Dspring.cloud.config.label="${spring_config_label_env}" -Dspring.profiles.active="${active_profile_env}" -Dspring.cloud.config.uri="${spring_config_url_env}" certify-service.jar ; \
```

Comment:
should we also pass the "spring.application.name" to change property files based on env ?

Reply:
this can be done through docker compose env variable `SPRING_CONFIG_NAME`